src/Rules/StaticAssertOverThisAndStaticRule.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ikvasnica\PHPStan\Rules;
+
+use PhpParser\Node;
+use PhpParser\NodeAbstract;
+use PHPStan\Analyser\Scope;
+use PHPStan\Rules\Rule;
+use PHPUnit\Framework\Assert;
+
+/**
+ * @implements \PHPStan\Rules\Rule<\PhpParser\NodeAbstract>
+ */
+final class StaticAssertOverThisAndStaticRule implements Rule
+{
+    public function getNodeType(): string
+    {
+        return NodeAbstract::class;
+    }
+
+    /**
+     * @return array<string>
+     */
+    public function processNode(Node $node, Scope $scope): array
+    {
+        if (! TestRuleHelper::isCallToAssert($node, $scope)) {
```

Comment:
Je nejaky duvod, ze je tahle detekce v helper tride? Nestacilo by to jako privatni metoda?

Reply:
Uz to vidim, pouziva se to i v jinem pravidle. Tak co jinak: co to dat do nejake specificke tridy, treba CallToAssertDetector, a nemit to v obecnem helperu (ktery bych cely rozebral na mensi  a specifictejsi casti)?

Reply:
Máš pravdu, dám do zvlášť třídy ;)